fearless/Common/Services/PayoutRewardsService/PayoutValidatorsForNominatorFactory.swift

```diff
@@ -5,225 +5,43 @@ import IrohaCrypto
-            dependencies: dependecies
-        )
-    }
-
-    private func createSubscanQueryOperation<T>(
-        address: String,
-        moduleName: String,
-        callName: String,
-        mapper: AnyMapper<JSON, T>,
-        reducer: AnyReducer<T, Set<AccountId>>
-    ) -> BaseOperation<Set<AccountId>> {
-        let url = subscanBaseURL
-            .appendingPathComponent(SubscanApi.extrinsics)
-
-        let params = SubscanQueryParams(address: address, moduleName: moduleName, callName: callName)
-
-        return SubscanQueryService(
-            url: url,
-            params: params,
-            mapper: mapper,
-            reducer: reducer,
-            initialResultValue: [],
-            subscanOperationFactory: subscanOperationFactory,
-            operationManager: operationManager
-        ).longrunOperation()
+        self.subqueryURL = subqueryURL
     }
 }
 
 extension PayoutValidatorsForNominatorFactory: PayoutValidatorsFactoryProtocol {
-    func createResolutionOperation(for address: AccountAddress) -> CompoundOperationWrapper<[AccountId]> {
-        let controllersQueryWrapper = createControllersQueryWrapper(nominatorStashAddress: address)
-
-        let validatorsOperation: BaseOperation<[[AccountId]]> = OperationCombiningService<[AccountId]>(
-            operationManager: operationManager
-        ) {
-            let controllers = try controllersQueryWrapper.targetOperation.extractNoCancellableResultData()
-            let wrapper = try self.createValidatorsQueryWrapper(controllers: controllers)
-
-            return [wrapper]
-        }.longrunOperation()
-
-        validatorsOperation.addDependency(controllersQueryWrapper.targetOperation)
-
-        let mapOperation = ClosureOperation<[AccountId]> {
-            try validatorsOperation.extractNoCancellableResultData().flatMap { $0 }
+    func createResolutionOperation(
+        for address: AccountAddress,
+        dependingOn historyRangeOperation: BaseOperation<ChainHistoryRange>
+    ) -> CompoundOperationWrapper<[AccountId]> {
+        let source = SubqueryEraStakersInfoSource(url: subqueryURL, address: address)
+        let operation = source.fetch {
+            try? historyRangeOperation.extractNoCancellableResultData()
         }
-
-        mapOperation.addDependency(validatorsOperation)
-        let dependencies = controllersQueryWrapper.allOperations + [validatorsOperation]
-
-        return CompoundOperationWrapper(targetOperation: mapOperation, dependencies: dependencies)
+        operation.addDependency(operations: [historyRangeOperation])
+
+        let mergeOperation = ClosureOperation<[AccountId]> {
+            let erasInfo = try operation.targetOperation.extractNoCancellableResultData()
+
+            let addressFactory = SS58AddressFactory()
+            return erasInfo
+                .compactMap { validatorInfo -> AccountAddress? in
+                    let contains = validatorInfo.others.contains(where: { $0.who == address })
```

Comment:
Why do we need this filter if we already assume to apply it passing an address to `SubqueryEraStakersInfoSource`? So erasInfo by definition must not contain validators for which `address` is not in `others`.